gguf-py/scripts/gguf_dump.py

```diff
@@ -254,16 +254,26 @@ def dump_markdown_metadata(reader: GGUFReader, args: argparse.Namespace) -> None
                 value = str(field.parts[-1][0])
         else:
             if field.types[0] == GGUFValueType.ARRAY:
+                displayed_values = 0
                 curr_type = field.types[1]
                 if curr_type == GGUFValueType.STRING:
                     render_element = min(5, total_elements)
                     for element_pos in range(render_element):
-                        value += repr(str(bytes(field.parts[-1 - element_pos]), encoding='utf-8')[:5]) + (", " if total_elements > 1 else "")
+                        truncate_length = 30
+                        value_string = repr(str(bytes(field.parts[-1 - (total_elements - element_pos - 1) * 2]), encoding='utf-8'))
```

Comment:
Are you sure the indexing is correct?

Let's say `total_elements` is `8`:

- when `element_pos == 0`: `-1 - (8 - 0 - 1) * 2` == `-1 - (7) * 2` == `-1 - 14` == `-15`
- when `element_pos == 1`: `-1 - (8 - 1 - 1) * 2` == `-1 - (6) * 2` == `-1 - 12` == `-13`

Reply:
The reason for jumping by two... is that there seem to be a 'length' specifier for each string there... I'm not familar how to destruct gguf array and this approach seems to work well enough. But happy to see your approach instead.